x-pack/plugins/security/public/views/management/roles_grid/components/roles_grid_page.tsx

```diff
@@ -0,0 +1,298 @@
+          }
+        </EuiPageContentBody>
+      </EuiPageContent>
+    );
+  };
+
+  private getColumnConfig = (intl: InjectedIntl) => {
+    const reservedRoleDesc = intl.formatMessage({
+      id: 'xpack.security.management.roles.reservedColumnDescription',
+      defaultMessage: 'Reserved roles are built-in and cannot be edited or removed.',
+    });
+
+    return [
+      {
+        field: 'name',
+        name: intl.formatMessage({
+          id: 'xpack.security.management.roles.nameColumnName',
+          defaultMessage: 'Role',
+        }),
+        sortable: true,
+        truncateText: true,
+        render: (name: string, record: Role) => {
+          return (
+            <EuiText color="subdued" size="s">
+              <EuiLink data-test-subj="roleRowName" href={getRoleManagementHref(name)}>
+                {name}
+              </EuiLink>
+              {!isRoleEnabled(record) && (
+                <FormattedMessage
+                  id="xpack.security.management.roles.disabledTooltip"
+                  defaultMessage=" (disabled)"
+                />
+              )}
+            </EuiText>
+          );
+        },
+      },
+      {
+        field: 'metadata._reserved',
+        name: (
+          <EuiToolTip content={reservedRoleDesc}>
+            <span className="rolesGridPage__reservedRoleTooltip">
+              <FormattedMessage
+                id="xpack.security.management.roles.reservedColumnName"
+                defaultMessage="Reserved"
+              />
+              <EuiIcon size="s" color="subdued" type="questionInCircle" className="eui-alignTop" />
+            </span>
+          </EuiToolTip>
+        ),
+        sortable: true,
+        dataType: 'boolean',
+        align: 'right',
+        description: reservedRoleDesc,
+        render: (reserved: boolean | undefined) => {
+          const label = intl.formatMessage({
+            id: 'xpack.security.management.roles.reservedRoleIconLabel',
+            defaultMessage: 'Reserved role',
+          });
+
+          return reserved ? (
+            <span title={label}>
+              <EuiIcon aria-label={label} data-test-subj="reservedRole" type="check" />
+            </span>
+          ) : null;
+        },
+      },
+    ];
+  };
+
+  private getVisibleRoles = () => {
+    const { roles, filter } = this.state;
+
+    return filter
+      ? roles.filter(({ name }) => {
+          const normalized = `${name}`.toLowerCase();
+          const normalizedQuery = filter.toLowerCase();
+          return normalized.indexOf(normalizedQuery) !== -1;
+        })
+      : roles;
+  };
+
+  private handleDelete = () => {
+    this.setState({
+      selection: [],
+      showDeleteConfirmation: false,
+    });
+    this.loadRoles();
+  };
+
+  private async loadRoles() {
+    try {
+      const roles = await RolesApi.getRoles();
+      // ensure each role has a `metadata._reserved` property to enable proper sorting.
+      const transformedRoles = roles.map(role => ({
```

Comment:
Instead of doing this, we can change line 204 to the following:
```
sortable: ({ metadata }: any) => (metadata && metadata._reserved) || false,
```

Reply:
TIL, thanks! I didn't see that in the EUI docs